lib/src/core/metadata/lifecycle_hooks.dart

```diff
@@ -77,75 +77,33 @@ abstract class OnInit {
 /// Implement this interface to override the default change detection algorithm
 /// for your directive.
 ///
-/// `ngDoCheck` gets called to check the changes in the directives instead of
+/// [ngDoCheck] gets called to check the changes in the directives instead of
 /// the default algorithm.
 ///
 /// The default change detection algorithm looks for differences by comparing
 /// bound-property values by reference across change detection runs. When
-/// `DoCheck` is implemented, the default algorithm is disabled and `ngDoCheck`
+/// [DoCheck] is implemented, the default algorithm is disabled and [ngDoCheck]
 /// is responsible for checking for changes.
 ///
 /// Implementing this interface allows improving performance by using insights
 /// about the component, its implementation and data types of its properties.
 ///
-/// Note that a directive should not implement both `DoCheck` and [OnChanges] at
-/// the same time.  `ngOnChanges` would not be called when a directive
-/// implements `DoCheck`. Reaction to the changes have to be handled from within
-/// the `ngDoCheck` callback.
+/// Note that a directive should not implement both [DoCheck] and [OnChanges] at
+/// the same time.  [ngOnChanges] would not be called when a directive
+/// implements [DoCheck]. Reaction to the changes have to be handled from within
+/// the [ngDoCheck] callback.
```

Comment:
Did you verify this. I vaguely remember discussions in TS that this is not actually the case.
Seems I even started it myself :D https://github.com/angular/angular/issues/6810